src/compiler/checker.ts

```diff
@@ -6662,46 +6677,95 @@ namespace ts {
             return s.valueDeclaration ? s.valueDeclaration.kind : SyntaxKind.PropertyDeclaration;
         }
 
-        function getDeclarationFlagsFromSymbol(s: Symbol) {
+        function getDeclarationFlagsFromSymbol(s: Symbol): NodeFlags {
             return s.valueDeclaration ? getCombinedNodeFlags(s.valueDeclaration) : s.flags & SymbolFlags.Prototype ? NodeFlags.Public | NodeFlags.Static : 0;

```

Comment:
It seems this is an issue that showed up elsewhere. I'll defer making any changes here (except for keeping the type annotation -- it was helpful for reading/writing the rest of the code).